### PR TITLE
Remove duplicated statistics tests in parquet

### LINKED ISSUE #6185: Remove test duplication in parquet statistics tets
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

When adding additional support to the statistics converter as @Kev1n8 did in https://github.com/apache/arrow-rs/pull/6181 the presence of two sets of tests is quite confusing about where to add tests for the new test

There are tests in both
* https://github.com/apache/arrow-rs/blob/master/parquet/src/arrow/arrow_reader/statistics.rs
* https://github.com/apache/arrow-rs/blob/master/parquet/tests/arrow_reader/statistics.rs

**Describe the solution you'd like**
I would like to remove the rundant tests 

**Describe alternatives you've considered**
I think the clearest alternateive is to move all the tests from https://github.com/apache/arrow-rs/blob/master/parquet/src/arrow/arrow_reader/statistics.rs to  https://github.com/apache/arrow-rs/blob/master/parquet/tests/arrow_reader/statistics.rs

We should only move the non duplicated tests.

So that means something like:
- [ ] Leave a comment with a note about where the tests are in https://github.com/apache/arrow-rs/blob/master/parquet/src/arrow/arrow_reader/statistics.rs
- [ ] Keep edge condition tests https://github.com/apache/arrow-rs/blob/f2de2cd3c8a8496e1f96e58b50786227bfe915f7/parquet/src/arrow/arrow_reader/statistics.rs#L1513-L1522
- [ ] Remove tests in https://github.com/apache/arrow-rs/blob/f2de2cd3c8a8496e1f96e58b50786227bfe915f7/parquet/src/arrow/arrow_reader/statistics.rs#L1548-L2113 that have corresponding coverage
- [ ] Keep data file validation: https://github.com/apache/arrow-rs/blob/f2de2cd3c8a8496e1f96e58b50786227bfe915f7/parquet/src/arrow/arrow_reader/statistics.rs#L2115-L2235
- [ ] Port fixed length decimal legacy: https://github.com/apache/arrow-rs/blob/f2de2cd3c8a8496e1f96e58b50786227bfe915f7/parquet/src/arrow/arrow_reader/statistics.rs#L2237-L2258

**Additional context**
This duplication is a historical artifact of how this code was developed in DataFusion and then it got brought over when @efredine  ported the work in https://github.com/apache/arrow-rs/pull/6046

### PR DESCRIPTION
**Pull request by Kev1n8**
# Which issue does this PR close?

Closes #6185

# Rationale for this change
 

# What changes are included in this PR?
Remove the duplicated tests, mainly the `roundtrip_xxx` tests.
Keep `roundtrip_empty` and make it further with more types than just `boolean`.

The data file tests could pass but it changed the testing and parquet-testing folders. I'm not sure if it's normal.

# Are there any user-facing changes?
No.

### CODE REVIEW

File: parquet/tests/arrow_reader/statistics.rs [hunk 1]
```diff
@@ -2195,3 +2195,429 @@ async fn test_column_non_existent() {
...
+        TestFile::new("nan_in_stats.parquet")
+            .with_column(ExpectedColumn {
+                name: "x",
+                expected_min: Arc::new(Float64Array::from(vec![Some(1.0)])),
+                expected_max: Arc::new(Float64Array::from(vec![Some(f64::NAN)])),
+            })
+            .run();
+    }
+
+    #[test]
```
Comment (alamb) [inline parquet/tests/arrow_reader/statistics.rs:2288]: 👍

**Review by alamb — APPROVED** [review]
Thank you @Kev1n8  🙏  this looks great

Thank you so much for helping to keep the code clean

I noticed that you also updated the `testing` (arrow-testing) and `parquet-testing` pin but they also seems like good changes to me (pick up the very latest testing files)
